### ISSUE TITLE
bundle install error

### ISSUE THREAD

**Issue by Wandimi** [instructure/canvas-lms#1807]
root@piu-HP-200-G3-AiO:/var/canvas# bundle install
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.

 

[!] There was an error evaluating `Gemfile`: ruby_version must match the:engine_version for MRI. Bundler cannot continue.

 

 #  from /var/canvas/Gemfile.d/_before.rb:39
 #  -------------------------------------------
 #   >   #  
 #  -------------------------------------------

**Comment by Captainguoz** [comment]
I also had this problem

**Comment by Wandimi** [comment]
> I also had this problem

right now an experiencing another error in the ruby version and the gem version installed not the same... its hectic

**Comment by Yetoo1** [comment]
Yeah it's a mess, what I did was uninstall bundle and bundler before uninstalling all ruby versions via apt or other means and downloaded and used ruby-install `2.6.6 -i /usr` to compile install ruby and friends to /usr/bin (it's confusing at first because you first expect it to install the binaries to the path specified such as /usr/bin, but it instead does /usr/bin/bin with the executables in the last bin so that's why you need /usr) and then you should be able to use sudo gem install bundle:2.2.0 or bundle:2.2.0 and then run bundle _2.2.0_ for the rest of the tasks.

**Comment by Wandimi** [comment]
I CANT RUN CANVAS ON BROWSER HAVING INSTALLED ON UBUNTU  18.04 VM SERVER

I have managed to install canvas successfully, my apache is running successfully. managed to edit sites-available/canvas.config but it cant load on my browser.
see below. 
<VirtualHost *:80>
  ServerName _mydomain.edu_
  ServerAlias _41.204.XXX.XXX_
  ServerAdmin _web.admin@edu_
  DocumentRoot /var/canvas/public
  RewriteEngine On
  RewriteCond %{HTTP:X-Forwarded-Proto} !=https
  RewriteCond %{REQUEST_URI} !^/health_check
  RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [L]
  ErrorLog /var/log/apache2/canvas_errors.log
  LogLevel warn
  CustomLog /var/log/apache2/canvas_access.log combined
  SetEnv RAILS_ENV production
  <Directory /var/canvas/public>
    Allow from all
    Options -MultiViews
  </Directory>
</VirtualHost>
<VirtualHost *:443>
  ServerName _mydomain.edu_
  ServerAlias _41.204.XXX.XXX_
  ServerAdmin _web.admin@edu_
  DocumentRoot /var/canvas/public
  ErrorLog /var/log/apache2/canvas_errors.log
  <Directory /var/canvas/public>
    Allow from all
    Options -MultiViews
  </Directory>
</VirtualHost>

**### **UPON RUNNING MY DOMAIN THE ERROR DISPLAYS**** 

This site can’t be reached
_mydomain.edu_ refused to connect.
Try:

Checking the connection
Checking the proxy and the firewall
ERR_CONNECTION_REFUSED

**WHERE AM I WRONG. KINDLY HELP**

**Comment by Wandimi** [comment]
I  managed this problem, Just a kind request if you can help me out, am stuck after installation, (my canvas is loading from Server I.P on port 3000, I have tried to redirect it to load from my Domain but no successful, am almost giving up) I have been using the product documentation all through. if you know the process kindly help out, reach me on elijahwandimi@gmail.com

**Comment by Wandimi** [comment]
@Yetoo1, @Captainguoz @canadaduane @treybean, Thanks  I  managed this problem, Just a kind request if you can help me out, am stuck after installation, (my canvas is loading from Server I.P on port 3000, I have tried to redirect it to load from my Domain but no successful, am almost giving up) I have been using the product documentation all through. if you know the process kindly help out, reach me on elijahwandimi@gmail.com

**Comment by Yetoo1** [comment]
@Wandimi Are you following https://github.com/instructure/canvas-lms/wiki/Production-Start or https://github.com/instructure/canvas-lms/wiki/Quick-Start and edited config/domain.yml for respective production or development? Have you opened the ports on your router/switch?

**Comment by ccutrer** [comment]
Please use [the mailing list](https://groups.google.com/g/canvas-lms-users) for installation and operational issues.
